timing/timing_individual_function.py

```diff
@@ -1,110 +1,135 @@
-import time
+"""
+Performance analysis of parallel and NetworkX implementations of the input function
+"""
 
+from time import perf_counter
+import random
+import types
 import networkx as nx
 import pandas as pd
 import seaborn as sns
 from matplotlib import pyplot as plt
-
 import nx_parallel as nxp
+import joblib
+
+# Default Config
+joblib.parallel_config(n_jobs=-1)
+# To use NetworkX's parallel backend directly, uncomment:
+# nx.config.backends.parallel.active = True
+# nx.config.backends.parallel.n_jobs = 6
 
-# Code to create README heatmaps for individual function currFun
 heatmapDF = pd.DataFrame()
-# for bipartite graphs
-# n = [50, 100, 200, 400]
-# m = [25, 50, 100, 200]
-number_of_nodes_list = [200, 400, 800, 1600]
+number_of_nodes = [10, 50, 100, 300, 500]
+tournament_funcs = ["is_reachable", "tournament_is_strongly_connected"]
+bipartite_funcs = ["node_redundancy"]
 weighted = False
-pList = [1, 0.8, 0.6, 0.4, 0.2]
-currFun = nx.tournament.is_reachable
-"""
-for p in pList:
-    for num in range(len(number_of_nodes_list)):
-        # create original and parallel graphs
-        G = nx.fast_gnp_random_graph(
-            number_of_nodes_list[num], p, seed=42, directed=True
-        )
+edge_prob = [1, 0.8, 0.6, 0.4, 0.2]
+currFun = nx.algorithms.centrality.betweenness.betweenness_centrality
 
+if currFun.__name__ not in tournament_funcs:
+    for p in edge_prob:
+        for num in range(len(number_of_nodes)):
+            print(number_of_nodes[num])
+            # create original and parallel graphs
+            G = nx.fast_gnp_random_graph(
+                number_of_nodes[num], p, seed=42, directed=True
+            )
 
-        # for bipartite.node_redundancy
-        G = nx.bipartite.random_graph(n[num], m[num], p, seed=42, directed=True)
-        for i in G.nodes:
-            l = list(G.neighbors(i))
-            if len(l) == 0:
-                v = random.choice(list(G.nodes) - [i,])
-                G.add_edge(i, v)
-                G.add_edge(i, random.choice([node for node in G.nodes if node != i]))
-            elif len(l) == 1:
-                G.add_edge(i, random.choice([node for node in G.nodes if node != i and node not in list(G.neighbors(i))]))
+            # for bipartite graphs
+            if currFun.__name__ in bipartite_funcs:
+                n = [50, 100, 200, 400]
+                m = [25, 50, 100, 200]
+                G = nx.bipartite.random_graph(n[num], m[num], p, seed=42, directed=True)
+                for i in G.nodes:
+                    neighbors = list(G.neighbors(i))
+                    if len(neighbors) == 0:
+                        v = random.choice(
+                            list(G.nodes)
+                            - [
+                                i,
+                            ]
+                        )
```

Comment:
Subtraction of lists doesn't work in python. Is this code actually being executed?
Instead use sets, or a list comprehension something like 
```
                        v = random.choice([node for node in G if node != i])
```